README.md

```diff
@@ -8,6 +8,86 @@ on-chain details of the API3 DAO, including:
+
+The services are:
+
+```
+(end user) -> Cloudflare -> EC2 IP -> traefik (load balancer) -> api3-tracker (container)
+```
+
+Containers:
+- api3tracker: The FE and BE-service
+- postgres: The database the FE and BE rely on
+- traefik: A load balancer that encrypts HTTP responses (using the CF origin server key pair)
+- postgres-exporter: a service that exports the database as a backup on an interval
+
+Host services:
+The host OS also runs some cron services, these are:
+```bash
+*/10 * * * * root cd /home/ubuntu/src/github.com/api3dao/api3-tracker/terraform/workspaces/api3tracker-prod && ./bin/job_logs_download.sh >> /var/log/api3-logs-download.log 2>&1
+15,45 * * * * root cd /home/ubuntu/src/github.com/api3dao/api3-tracker/terraform/workspaces/api3tracker-prod && ./bin/job_supply_download.sh >> /var/log/api3-supply-download.log 2>&1
+0 * * * * root cd /home/ubuntu/src/github.com/api3dao/api3-tracker/terraform/workspaces/api3tracker-prod && ./bin/job_treasuries_download.sh >> /var/log/api3-treasuries-download.log 2>&1
+2,12,22,32,42,52 * * * * root cd /home/ubuntu/src/github.com/api3dao/api3-tracker/terraform/workspaces/api3tracker-prod && ./bin/job_state_update.sh >> /var/log/api3-state-update.log 2>&1
+10 0 * * * root cd /home/ubuntu/src/github.com/api3dao/api3-tracker/terraform/workspaces/api3tracker-prod && ./bin/job_shares_download.sh --tag . > /var/log/api3-shares-download.log 2>&1
+24 4 * * */3 root cd /home/ubuntu/src/github.com/api3dao/api3-tracker/terraform/workspaces/api3tracker-prod && bash ./bin/postgres-backup.sh >> /var/log/postgres-backups.log 2>&1
+```
+
+## Local developement using Docker
+Developers can run some or all services locally using Docker Swarm, or even bare-bones, without containerisation.
+
+One combination is running just postgres locally using Docker, eg:
+```bash
+docker run --rm -ti -p 5432:5432 postgres:15
+```
+and then running the FE and BE services directly (refer to Cron jobs below and `yarn next dev` in `package.json`).
+
+Alternatively, one can run services using Docker Swarm, but this lacks hot-reloading.
+
+### Local development using Docker Swarm
+If you haven't already enabled Swarm mode on your Docker instance, do so now (only has to be done once):
+```bash
+docker swarm init
+```
+The result of the above command can be ignored.
+
+Build the FE/BE image:
+```bash
+docker build -t api3dao/api3-tracker:latest .
+```
+
+Run the stack:
+```bash
+docker stack deploy -c dev-tools/docker-compose.yml tracker-stack
+```
+
+Some commands for visualising the services:
+```bash
+docker ps # all docker containers
+docker service ls # all swarm services
+docker service ps tracker-stack_postgres --no-trunc # show status of postgres
+docker stack rm tracker-stack # tear down the stack
+```
+
+Initialise the DB:
+```bash
+DATABASE_URL="postgres://postgres:postgres@127.0.0.1:5432/postgres?sslmode=disable" yarn prisma migrate deploy
+```
+
```

Comment:
Maybe add here about the application being available at `http://localhost:3000/` ?

And then perhaps delete the line below that mentions: "Default local environment starts website at [http://localhost:7040](http://localhost:7040)"? And actually - is that whole section even relevant anymore?

Reply:
🤷 
The problem is that there are two systems involved; Terraform and Docker.
You could argue there are even more involved (AWS), but that's another story.

I think the Terraform instructions mix TF installation for the purposes of deployment to AWS vs TF installation for deplying and/or testing on the local machine.

Generally I would avoid running TF against the local machine - and in fact it possibly won't even run correctly on Mac OS. One one hand I feel that section can be removed, but on the other it's possibly a useful reference as it is how the TF side can be tested locally. Perhaps just the section heading should be updated.